src/fsharp/MethodOverrides.fs

```diff
@@ -255,23 +255,40 @@ module DispatchSlotChecking =
 
         let isReqdTyInterface = isInterfaceTy g reqdTy 
         let showMissingMethodsAndRaiseErrors = (isReqdTyInterface || not isOverallTyAbstract)
-        let res = ref true
-        let fail exn = (res := false ; if showMissingMethodsAndRaiseErrors then errorR exn)
         
+        let mutable res = true
+        let fail exn =
+            res <- false
+            if showMissingMethodsAndRaiseErrors then
+                errorR exn
+
+        let rec listSameLength l1 l2 = 
+            match l1, l2 with
+            | [], [] -> true
+            | [], _ | _, [] -> false
+            | _ :: t1, _ :: t2 -> listSameLength t1 t2
+
+        let inline formatMethodInfoSignature dispatchSlot = 
```

Comment:
Don't use `inline`.  Indeed don't define this local function, it makes code harder to understand to be honest.